client/typescript/fine-uploader.d.ts

```diff
@@ -2984,397 +2315,1240 @@ declare namespace FineUploader {
+            /**
+             * Enables or disables cross-domain ajax calls (if the `expected` property is true) in IE9 and older.
+             *
+             * @default `true`
+             */
+            allowXdr: boolean;
+        }
+
+        /**
+         * AzureBlobPropertyNameFunction
+         */
+        export interface AzureBlobPropertyNameFunction {
+            constructor(id: number);
+        }
+
+        /**
+         * AzureBlobPropertyOptions
+         */
+        export interface AzureBlobPropertyOptions {
+            /**
+             * Describes the blob name used to identify the file in your Azure Blob Storage container.
+             *
+             * Possible values are
+             * * `'uuid'`
+             * * `'filename'`
+             * * `function`
+             *
+             * If the value is a function, Fine Uploader Azure will pass the associated file ID as a parameter when invoking your function.
+             * If the value is a function it may return one of a `qq.Promise` or a `String`
+             *
+             * @default `'uuid'`
+             */
+            name?: string | AzureBlobPropertyNameFunction;
+        }
+
+        /**
+         * AzureRequestOptions
+         */
+        export interface AzureRequestOptions extends RequestOptions {
+            /**
+             * URL for your Azure Blob Storage container
+             *
+             * @default `null`
+             */
+            containerUrl?: string;
+            /**
+             * Parameters passed along with each upload request.
+             *
+             * @default `{}`
+             */
+            params?: any;
+            /**
+             * Part of the parameter name that contains the name of the associated file which may differ from the blob name.
+             * Prefixed with 'x-ms-meta-' by Fine Uploader
+             *
+             * @default `'qqfilename'`
+             */
+            filenameParam?: string;
+        }
+
+        /**
+         * type for Azure's customHeaders function
+         */
+        export interface AzureCustomHeaderFunction {
+            constructor(id: number);
+        }
 
         /**
-         * Modify the endpoint that Fine Uploader should POST to when a file has been successfully uploaded to Azure Blob Storage.
-         * 
-         * @param string endpoint : An endpoint that Fine Uploader should POST to when a file has been successfully uploaded to Azure Blob Storage
-         * @param number id : An ID corresponding to a file
+         * AzureSignatureOptions
          */
-        setUploadSuccessEndpoint(endpoint: string, id?: number): void;
+        export interface AzureSignatureOptions {
+            /**
+             * Additional headers sent along with each signature request.
+             *
+             * If you  a function as the value, the associated file's ID will be passed to your function when it is invoked
```

Comment:
@jclangst I guess since you did replace all for all occurrences of word `declare`, it has led to sentences like those. 
see `declare` missing between `If you` and the rest of the sentence